File: src/main/java/com/baulsupp/oksocial/network/IPvMode.java
package com.baulsupp.oksocial.network;

public enum IPvMode {
  SYSTEM,
  IPV6_FIRST,
  IPV4_FIRST,
  IPV6_ONLY,
  IPV4_ONLY;

  public static IPvMode fromString(String ipMode) {
    switch (ipMode) {
      case "ipv6":
        return IPvMode.IPV6_FIRST;
      case "ipv4":
        return IPvMode.IPV4_FIRST;
      case "ipv6only":
        return IPvMode.IPV6_ONLY;
      case "ipv4only":
        return IPvMode.IPV4_ONLY;
      default:
        return IPvMode.SYSTEM;
    }
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/network/IPvMode.kt
package com.baulsupp.oksocial.network

import com.baulsupp.oksocial.output.UsageException

enum class IPvMode(val code: String) {
  SYSTEM("system"),
  IPV6_FIRST("ipv6"),
  IPV4_FIRST("ipv4"),
  IPV6_ONLY("ipv6only"),
  IPV4_ONLY("ipv4only");

  companion object {
    @JvmStatic
    fun fromString(ipMode: String): IPvMode =
      IPvMode.values().find { it.code == ipMode } ?: throw UsageException("Unknown value $ipMode")
  }
}
